src/Plugin/LoggerSubscriber.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace OpenEuropa\EPoetry\Plugin;
+
+use Phpro\SoapClient\Event\FaultEvent;
+use Phpro\SoapClient\Event\RequestEvent;
+use Phpro\SoapClient\Event\ResponseEvent;
+use Phpro\SoapClient\Events;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Class ePoetryLogPlugin.
+ */
+class LoggerSubscriber implements EventSubscriberInterface
```

Comment:
This class does two things:

1. It subscribes to the library event dispatcher to log events fired withing it
2. It actually logs them

Now, we should divide this class into two:

1. A simple subscriber that just subscribe to events and calls a (new) logger class
2. The above mentioned logger class that will encapsulate how the library logs things, like the text format it sues, the log levels etc.

If we do this then other classes, such as middlewares (\cc @dxvargas) can use the logger and have things properly logged in consistent way without needing to fire events, etc.